packages/synapse-react-client/src/components/EntityHeaderTable/EntityHeaderTable.tsx

```diff
@@ -305,11 +305,13 @@ export const EntityHeaderTable = (props: EntityHeaderTableProps) => {
         table.setColumnFilters([])
       }
     },
-    [showFilterControls],
+    [table, showFilterControls],
   )
 
   if (isLoading) {
-    return <SkeletonTable numCols={3} numRows={10} />
+    return (
+      <SkeletonTable numCols={3} numRows={Math.min(10, refsInState.length)} />
```

Comment:
If we're fetching < 10 entity headers, the skeleton table should render as many rows.